### PR TITLE
Support multiple COS instances scenario

### PR DESCRIPTION
**Pull request by akshaykumar-vijapur**
The following are the scenarios.

1. If user does not have any COS instance, Then toolchain will create a COS instance in `Lite Plan` with name as toolchain name and use it for further bucket creation.
2. if user has only one Instance, Then we assume that user is on `Lite Plan`. We will not create a new instance but re-use the same instance for the bucket creation.
3.  if user has more then Instance, Then we will create a COS instance with the name of the instance will be toolchain name and use it for further bucket creation.

### Description

---

### Testing Done

---

### CODE REVIEW

File: .build_pipeline/task.yaml [hunk 1]
```diff
@@ -182,13 +182,22 @@ spec:
...
+              echo "${INSTANCENAME} will be used to create the bucket."
+            elif [ "$COS_INSTANCE_COUNT" -eq "0" ]; then
+              INSTANCENAME=$COSBUCKETNAME
+              {
+              echo "Cloud Object Storage Instance does not exist. Proceeding with creation of instance named "$INSTANCENAME" with Lite plan.";
+              ibmcloud resource service-instance-create "$INSTANCENAME" cloud-object-storage lite global && echo "Instance creation successfull.."
+              } || {
+              echo "Failed to create an instance with Lite plan. Proceeding with creation of instance named "$INSTANCENAME" with Standard plan.";
+              ibmcloud resource service-instance-create "$INSTANCENAME" cloud-object-storage standard global && echo "Instance creation successfull.."
+              }
```
Comment (mundrapiyush36) [inline .build_pipeline/task.yaml]: @akshaykumar-vijapur Just got confused with the || condition here. What's going on in this code block ?

File: .build_pipeline/task.yaml [hunk 2]
```diff
@@ -182,13 +182,22 @@ spec:
...
             else
-              echo "Cloud Object Storage Instance does not exist. Proceeding with creation."
               INSTANCENAME=$COSBUCKETNAME
-              ibmcloud resource service-instance-create "$INSTANCENAME" cloud-object-storage lite global
-            fi                
+              echo "Multiple Cloud object storage instance detected. Proceeding with creation of instance named ${INSTANCENAME} with standard plan."
+              ibmcloud resource service-instance-create "$INSTANCENAME" cloud-object-storage standard global
+            fi
```
Comment (mundrapiyush36) [inline .build_pipeline/task.yaml:202]: Proceeding with the creation of Cloud Object Storage Instance with name ${INSTANCENAME} under Standard Plan.

**Review by mundrapiyush36 — CHANGES_REQUESTED** [review]

File: .build_pipeline/task.yaml (hunk 1 above)
Comment (akshaykumar-vijapur) [reply .build_pipeline/task.yaml]: @mundrapiyush36  It acts like a `try catch` block. Seems like we cannot find if a user is on lite/standard plan with `CLI`. So we will try to create a instance with `lite` plan,  if it works fine, else we will proceed with creation of instance with `standard` account.

File: .build_pipeline/task.yaml [hunk 3]
```diff
@@ -182,13 +182,22 @@ spec:
...
             if [ "$COS_INSTANCE_COUNT" -eq "1" ]; then
-              echo "Cloud Object Storage Instance already exists."
-              INSTANCENAME=$(ibmcloud resource service-instances --output JSON  --service-name cloud-object-storage | jq .[0] | jq -r .name)
+              INSTANCENAME=$(ibmcloud resource service-instances --output JSON --service-name cloud-object-storage | jq .[0] | jq -r .name)
+              echo "${INSTANCENAME} will be used to create the bucket."
+            elif [ "$COS_INSTANCE_COUNT" -eq "0" ]; then
+              INSTANCENAME=$COSBUCKETNAME
+              {
+              echo "Cloud Object Storage Instance does not exist. Proceeding with creation of Cloud Object Storage Instance named "$INSTANCENAME" under Lite plan.";
+              ibmcloud resource service-instance-create "$INSTANCENAME" cloud-object-storage lite global && echo "Instance creation successfull.."
```
Comment (mundrapiyush36) [inline .build_pipeline/task.yaml:193]: Instance creation successful.

File: .build_pipeline/task.yaml [hunk 4]
```diff
@@ -182,13 +182,22 @@ spec:
...
+              } || {
+              echo "Failed to create an instance with Lite plan. Proceeding with creation of Cloud Object Storage Instance named "$INSTANCENAME" under Standard plan.";
+              ibmcloud resource service-instance-create "$INSTANCENAME" cloud-object-storage standard global && echo "Instance creation successfull.."
```
Comment (mundrapiyush36) [inline .build_pipeline/task.yaml:196]: Instance creation successful.

**Review by mundrapiyush36 — APPROVED** [review]

**Pull request merged**